### PR TITLE
.github: Add release issue template

### PR DESCRIPTION
**Pull request by joestringer**
This release template only includes the list for micro / patch releases.
For full releases please consult the documentation.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10452)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - [ ] [kubeadm]
+  - [ ] [kops]
+  - [ ] [kubespray]
+
+
+[release blockers]: https://github.com/cilium/cilium/labels/priority%2Frelease-blocker
+[backport PRs]: https://github.com/cilium/cilium/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+backports
+[new project]: https://github.com/cilium/cilium/projects/new
+[Cilium release-notes tool]: https://github.com/cilium/release
+[Docker Hub]:
```
Comment (joestringer) [inline .github/ISSUE_TEMPLATE/release_template.md]: I didn't find a generic link for this yet, need to fix this up.

**Comment by joestringer** [comment]
Looking at the template, there's a bunch of stuff we could fold into a proper make target. We could also potentially make use of github CLI to create the PRs and so on.

But for now I found this template useful to track items to check off during release.

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 1 above)
Comment (aanm) [reply .github/ISSUE_TEMPLATE/release_template.md]: What's the docker hub section supposed to be used for?

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Pre-release
+
+- [ ] Check that there are no [release blockers] for the targeted release version
+- [ ] Ensure that outstanding [backport PRs] are merged
+- [ ] Create a [new project] for the next release version
+  - [ ] Move any unresolved issues/PRs into the newly created release project
+- [ ] Push a PR including the changes necessary for the new release:
+  - [ ] Update the VERSION file to represent X.Y.Z
+  - [ ] Update helm charts via `make -C install/kubernetes`
```
Comment (aanm) [inline .github/ISSUE_TEMPLATE/release_template.md:18]: What about creating the helm charts for https://github.com/cilium/charts?

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 1 above)
Comment (joestringer) [reply .github/ISSUE_TEMPLATE/release_template.md]: Easily clicking to add the image build targets for the new release

**Pull request merged**